xcube/util/jsonschema.py

```diff
@@ -175,54 +206,91 @@ def to_dict(self) -> Dict[str, Any]:
-
-    @staticmethod
-    def _is_valid_datetime_or_date(instance: str):
+    # noinspection PyShadowingBuiltins
+    @classmethod
+    def _is_valid_value(cls, value: str, format: str):
         try:
-            jsonschema.validate(
-                instance=instance,
-                schema=dict(anyOf=[dict(type='string', format='date-time'),
-                                   dict(type='string', format='date')]),
-                format_checker=jsonschema.draft7_format_checker)
+            jsonschema.validate(value,
+                                dict(type='string', format=format),
+                                format_checker=jsonschema.draft7_format_checker)
             return True
         except jsonschema.ValidationError:
             return False
 
-    @staticmethod
-    def new_datetime_range() -> 'JsonArraySchema':
-        """Return a schema for a two-element array of dates or date-times"""
-        return JsonArraySchema(items=[
-                    JsonDatetimeSchema(any_of=[
-                        JsonDatetimeSchema(format='date-time', nullable=True),
-                        JsonDatetimeSchema(format='date', nullable=True),
-                    ]),
-                    JsonDatetimeSchema(any_of=[
-                        JsonDatetimeSchema(format='date-time', nullable=True),
-                        JsonDatetimeSchema(format='date', nullable=True),
-                    ], nullable=True)])
+
+class JsonDateSchema(JsonStringSchema, JsonDateAndTimeSchemaBase):
+    """
+    JSON schema for date instances.
+
+    :param min_date: optional minimum date.
+    :param max_date: optional maximum date.
+    """
+
+    # noinspection PyShadowingBuiltins
+    def __init__(self,
+                 min_date: str = None,
+                 max_date: str = None,
+                 **kwargs):
+        super().__init__(**kwargs, format='date')
+        self._validate_value(min_date, 'min_date', 'date')
+        self._validate_value(max_date, 'max_date', 'date')
+        self.min_date = min_date
+        self.max_date = max_date
+
+    def to_dict(self) -> Dict[str, Any]:
+        d = super().to_dict()
+        if self.min_date is not None:
+            d.update(minDate=self.min_date)
+        if self.max_date is not None:
+            d.update(maxDate=self.max_date)
+        return d
+
+    @classmethod
+    def new_range(cls,
+                  min_date: str = None,
+                  max_date: str = None,
+                  nullable: bool = False) -> 'JsonArraySchema':
+        """
+        Return a schema for a date range.
+        :param min_date: optional minimum date.
+        :param max_date: optional maximum date.
+        :param nullable: whether the whole range as well as individual start and end may be None.
+        :return: a JsonArraySchema with two items.
+        """
+        return JsonArraySchema(
+            items=[
+                JsonDateSchema(min_date=min_date, max_date=max_date, nullable=nullable),
```

Comment:
```suggestion
                JsonDateSchema(min_date=min_date, max_date=max_date, nullable=True),
```

The documentation says that `min_date` is optional, so its schema should be nullable whether or not the enclosing array schema is nullable.

Reply:
> The documentation says that min_date is optional, so its schema should be nullable whether or not the enclosing array schema is nullable.

Why do you think nullable depends on min_date ?